sql/rowexec/dml.go

```diff
@@ -54,7 +54,28 @@ func (b *BaseBuilder) buildInsertInto(ctx *sql.Context, ii *plan.InsertInto, row
 		return nil, err
 	}
 
+	var unlocker func()
 	insertExpressions := getInsertExpressions(ii.Source)
+	if ii.HasUnspecifiedAutoInc {
+		_, i, _ := sql.SystemVariables.GetGlobal("innodb_autoinc_lock_mode")
+		lockMode := i.(int64)
+		if err != nil {
+			return nil, err
+		}
```

Comment:
There's an `error` being swallowed here. You've got the check for it, just not assigning to it.

Reply:
Fixed.